modules/graphics/src/font.cxx

```diff
@@ -7,82 +7,45 @@
   {
-    shader = NULL;
-    m_atlas = texture_atlas_new(512, 512, 1);
-    m_font = texture_font_new_from_file(m_atlas, font_size, filename.c_str());
-    texture_font_load_glyphs(m_font, L" !\"#$%&'()*+,-./0123456789:;<=>?"
+    atlas = texture_atlas_new(512, 512, 1);
+    font = texture_font_new_from_file(atlas, font_size, filename.c_str());
+    texture_font_load_glyphs(font, L" !\"#$%&'()*+,-./0123456789:;<=>?"
                                    L"@ABCDEFGHIJKLMNOPQRSTUVWXYZ[\\]^_"
                                    L"`abcdefghijklmnopqrstuvwxyz{|}~");
   }
 
   Font::~Font() {
-    texture_font_delete(m_font);
-    texture_atlas_delete(m_atlas);
-  }
-
-  void Font::DrawText(const float x, const float y, const float z,
-                      const Moon::String &text) {
-    DrawText(x, y, z, text, DefaultColor);
-  }
-
-  void Font::DrawText(const float x, const float y, const float z,
-                      const Moon::String &text, const Vector4 &color) {
-    RenderState render_ops;
-    render_ops.color = color;
-    DrawText(x, y, z, text, render_ops);
+    texture_font_delete(font);
+    texture_atlas_delete(atlas);
   }
 
-  void Font::DrawText(const float x, const float y, const float z,
-                      const Moon::String &text, const RenderState &render_ops) {
-    if (!shader) return;
-    // outline
-    if (render_ops.outline > 0) {
-      m_font->outline_type = 2;
-      m_font->outline_thickness = render_ops.outline;
-      AddText(text, render_ops.outline_color);
-    }
-    m_font->outline_type = 0;
-    m_font->outline_thickness = 0;
-    AddText(text, render_ops.color);
-
-    shader->Use();
-    //model matrix
-    glm::mat4 model_matrix = glm::rotate( // rotate it for 180 around the x-axis, because the text was upside down
-      glm::translate(render_ops.transform, glm::vec3(x, y + m_font->ascender, z)), // move it to the correct position in the world
-      glm::radians(180.0f),
-      glm::vec3(1.0f, 0.0f, 0.0f)
-    );
-    // calculate the ModelViewProjection matrix (faster to do on CPU, once for all vertices instead of per vertex)
-    glm::mat4 mvp_matrix = Shader::projection_matrix * Shader::view_matrix * model_matrix;
-    shader->SetUniform("mvp_matrix", mvp_matrix);
-
-    // Set texture ID
-    glBindTexture(GL_TEXTURE_2D, m_atlas->id);
-    shader->SetUniform("tex", /*GL_TEXTURE*/0);
-    m_buffer.Render(GL_TRIANGLES);
-    m_buffer.Clear();
-  }
-
-  void Font::AddText(const Moon::String &text, const Vector4 &c) {
+  void Font::FillTextBuffer(VertexBuffer *vbo, const String &text, const Vector4 &c, const float line_height) {
+    const float line_space = line_height * font->size;
     float cursor = 0; // position of the write cursor
-
+    float line = 0;
     for(size_t i = 0; i < text.length(); ++i) {
-      texture_glyph_t *glyph = texture_font_get_glyph(m_font, text[i]);
-      if(glyph != NULL) {
+      if (isNewline(text[i])) {
+        cursor = 0;
+        line += line_space;
```

Comment:
:( true, you can align the entire chunk of text by just swiping its w/h and doing the appropriate calculations, but for aligning line by line, well I didn't think of that ;x